(Improvement) Request Model

99Anvar99 opened the pull request:
- We don't want to request models that does not exist! or invalid

99Anvar99 committed:
(Improvement) Request Model

- We don't want to request models that does not exist! or invalid

99Anvar99 committed:
Fix for issue#2670

Yimura commented on src/util/entity.cpp:
```diff
@@ -218,7 +219,15 @@ namespace big::entity
+        {
+			location.z = height;
+        }
+		else
```
Mixed indentation, run `clang-format -i src/**.hpp src/**.cpp` to fix.

Yimura commented on src/util/ped.hpp:
```diff
@@ -446,17 +446,20 @@ namespace big::ped
+			{
+				STREAMING::REQUEST_MODEL(hash);
+				script::get_current()->yield();
+			}
```
Since this piece of code is being duplicated quite a bit, how about making it a general purpose helper function?

99Anvar99 replied:
Sure thing.

Yimura reviewed (changes requested):
Just some mixed indentations here and there as well as duplication of commonly used code.

### To Do:
 - [x] Move the code block that checks if a model is valid and then loads it to a helper function
 - [x] Run clang-format to make the code that you submit uniform with the main code base.

99Anvar99 committed:
Created helper class for request model

### Additional Fixes
- Removed unreachable code in vehicle.cpp.
- Removed redundant code.
- Removed unused code.

Yimura commented on src/util/entity.cpp:
```diff
@@ -193,33 +193,67 @@ namespace big::entity
+					break;
+				STREAMING::REQUEST_MODEL(hash);
+				script::get_current()->yield();
+			}
```
`has_loaded` is never updated in this loop, might cause a deadlock.

Yimura commented on src/util/entity.cpp:
```diff
@@ -193,33 +193,67 @@ namespace big::entity
+			return true;
+		}
+
+		return false;
```
Can be changed to `return has_loaded`.

Yimura commented on src/util/world_model.hpp:
```diff
@@ -15,26 +15,21 @@ namespace big::world_model
+			const auto object = OBJECT::CREATE_OBJECT(hash, location.x, location.y, location.z, is_networked, false, false);
 
-		const auto object = OBJECT::CREATE_OBJECT(hash, location.x, location.y, location.z, is_networked, false, false);
+			memcpy(g_pointers->m_gta.m_world_model_spawn_bypass, backup.data(), patch_size);
```
I know this isn't your fault but could you update the patching of the `m_world_model_spawn_bypass` and make it into a `memory::byte_patch`?

Similarly to how it was implemented here:
https://github.com/YimMenu/YimMenu/blob/7efbee2e85ba2aee773e398cd9e66d54e640aca5/src/byte_patch_manager.cpp#L16

And here's an example usage:
https://github.com/YimMenu/YimMenu/blob/7efbee2e85ba2aee773e398cd9e66d54e640aca5/src/backend/looped/self/police.cpp#L19

99Anvar99 replied:
Yes, i will update it.

99Anvar99 committed:
Fixes and world spawn bypass

99Anvar99 committed:
Logic Fix

Yimura reviewed (approved):
Thanks for another PR fixing and improving some of the tech debt in YimMenu!
